docs/references/architecture/adr-117-multi-curve-support.md

```diff
@@ -0,0 +1,245 @@
+# ADR 117: Cryptography: Multi curve support
+
+## Change log
+
+* July 1st 2024: Initial proposal (Zondax AG: @raynaudoe @juliantoledano @jleni @educlerici-zondax @lucaslopezf)
+
+## Status
+
+DRAFT
+
+## Abstract
+
+This ADR proposes the implementation of [ADR-001-crypto-provider](https://github.com/cosmos/crypto/blob/main/docs/architecture/adr-001-crypto-provider.md). By refactoring the current `PrivValidator` implementations and replacing them with `CryptoProvider`s, we will allow for "pluggable cryptography" that enables a more modular design and future-proofs crypto tools.
+
+We will be extensively referencing concepts like CryptoProviders in this ADR. Please check the full ADR describing them [here](https://github.com/cosmos/crypto/blob/main/docs/architecture/adr-001-crypto-provider.md#crypto-provider).
+
+## Context
+
+The introduction of multi-curve support, by implementing the `CryptoProvider` interface and its components, in the CometBFT cryptographic package offers significant advantages. By not being restricted to a single cryptographic curve, developers can choose the most appropriate curve based on security, performance, and compatibility requirements. This flexibility enhances the application's ability to adapt to evolving security standards and optimizes performance for specific use cases, helping to future-proof the codebase's cryptographic capabilities.
+
+## Decision
+
+We will:
```

Comment:
I know you folks define `CryptoProvider` in [ADR-001-crypto-provider](https://github.com/cosmos/crypto/blob/main/docs/architecture/adr-001-crypto-provider.md), but for readability purposes, would you mind adding the `CryptoProvider` definition also here? Thanks!